src/mqtt_client/mod.rs

```diff
@@ -207,29 +205,30 @@ where
             .set_keepalive_interval(KEEPALIVE_INTERVAL_SECONDS)
             .unwrap();
 
+        let prefix = String::from(prefix);
+
         // Configure a will so that we can indicate whether or not we are connected.
-        let mut connection_topic: String<MAX_TOPIC_LENGTH> = String::from(prefix);
+        let mut connection_topic = prefix.clone();
         connection_topic.push_str("/alive").unwrap();
         mqtt.client()
             .set_will(
                 &connection_topic,
```

Comment:
Can this just be `&prefix`? The borrow should end before ownership of `prefix` moves I believe

Reply:
It needs the `/alive` appended.